homework-submission/react-with-json-server/src/App.js

```diff
@@ -0,0 +1,95 @@
+import React, { Component } from 'react';
+import Util from './utility.js';
+import Login from "./login-form.js";
+import AddingForm from "./add-form.js";
+import Photo from "./photo.js";
+
+
+class App extends Component {
+
+  state = {
+    userName: "no user",
+    login: false,
+    users: []
+  }
+
+  getContent = () =>{
+    Util.fetchJSON("http://localhost:4000/photos")
+      .then(users => this.setState({ users })).catch(err => console.error(err));
+  }
+
+  componentDidMount() { this.getContent() }
+
+  loginLogout = (e) => {
+    if (e.target) {
+      e.preventDefault();
+      this.setState({ userName: e.target.userName.value });
+    }
+    this.setState({login: !this.state.login});
+  }
+
+  handleSubmit = (e) => {
+    e.preventDefault();
+    let newPost = {
+     
+      photoUrl: e.target.url.value,
+      author: this.state.userName,
+      description: e.target.description.value,
+      comment: [
+        {
+          author: "jalal",
+          text: "I like the place"
+        }
+      ],
+      like: [
+        "jalal",
+        "karam"
+      ]
+    }
+    Util.postJSON("http://localhost:4000/photos", newPost)
+      .then(res => this.getContent()).catch(err => console.error(err));
```

Comment:
as last homework try to fine another way to refresh your DOM, 
it is really bad to fetch all data for every update